### ISSUE TITLE
Problem with JSON deserialization

### ISSUE THREAD

**Issue by tochka** [InstaSharp/InstaSharp#106]
I catch the exception when I collect posts by cats hashtag

Error message: `{"JSON integer 1572296036331324 is too large or small for an Int32. Path 'data[2].location.id', line 1, position 5454."}`

Stack trace:
```
   at Newtonsoft.Json.JsonTextReader.ParseNumber(ReadType readType)
   at Newtonsoft.Json.JsonTextReader.ReadNumberValue(ReadType readType)
   at Newtonsoft.Json.JsonTextReader.ReadAsInt32()
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.ReadForType(JsonReader reader, JsonContract contract, Boolean hasConverter)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.PopulateObject(Object newObject, JsonReader reader, JsonObjectContract contract, JsonProperty member, String id)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateObject(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateValueInternal(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.SetPropertyValue(JsonProperty property, JsonConverter propertyConverter, JsonContainerContract containerContract, JsonProperty containerProperty, JsonReader reader, Object target)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.PopulateObject(Object newObject, JsonReader reader, JsonObjectContract contract, JsonProperty member, String id)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateObject(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateValueInternal(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.PopulateList(IList list, JsonReader reader, JsonArrayContract contract, JsonProperty containerProperty, String id)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateList(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, Object existingValue, String id)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateValueInternal(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.SetPropertyValue(JsonProperty property, JsonConverter propertyConverter, JsonContainerContract containerContract, JsonProperty containerProperty, JsonReader reader, Object target)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.PopulateObject(Object newObject, JsonReader reader, JsonObjectContract contract, JsonProperty member, String id)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateObject(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateValueInternal(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.Deserialize(JsonReader reader, Type objectType, Boolean checkAdditionalContent)
   at Newtonsoft.Json.JsonSerializer.DeserializeInternal(JsonReader reader, Type objectType)
   at Newtonsoft.Json.JsonConvert.DeserializeObject(String value, Type type, JsonSerializerSettings settings)
   at Newtonsoft.Json.JsonConvert.DeserializeObject[T](String value, JsonSerializerSettings settings)
   at Newtonsoft.Json.JsonConvert.DeserializeObject[T](String value)
   at InstaSharp.Extensions.HttpClientExtensions.<ExecuteAsync>d__0`1.MoveNext()
```

**Comment by tochka** [comment]
Example:
```JSON
{
	"attribution": null,
	"tags":  [
		"cute",
		"babysavannah",
		"cuddlesesh",
		"lol",
		"bigvssmall",
		"cats",
		"kittycity",
		"lillulu"
	],
	"type": "image",
	"location":  {
		"latitude": 44.262281874977,
		"name": "267 Newburgh Rd",
		"longitude": -76.94864988327,
		"id": 897882256938993
	},
	"comments":  {
		"count": 0,
		"data":  []
	},
	"filter": "Normal",
	"created_time": "1462724530",
	"link": "https://www.instagram.com/p/BFJx31WPEJu/",
	"likes":  {
		"count": 0,
		"data":  []
	},
	"images":  {
		"low_resolution":  {
			"url": "https://scontent.cdninstagram.com/t51.2885-15/s320x320/e35/13129595_896964890448784_210397585_n.jpg?ig_cache_key=MTI0NTc0NjExMzA3NjI4MTk2Ng%3D%3D.2",
			"width": 320,
			"height": 320
		},
		"thumbnail":  {
			"url": "https://scontent.cdninstagram.com/t51.2885-15/s150x150/e35/13129595_896964890448784_210397585_n.jpg?ig_cache_key=MTI0NTc0NjExMzA3NjI4MTk2Ng%3D%3D.2",
			"width": 150,
			"height": 150
		},
		"standard_resolution":  {
			"url": "https://scontent.cdninstagram.com/t51.2885-15/s640x640/sh0.08/e35/13129595_896964890448784_210397585_n.jpg?ig_cache_key=MTI0NTc0NjExMzA3NjI4MTk2Ng%3D%3D.2",
			"width": 640,
			"height": 640
		}
	},
	"users_in_photo":  [],
	"caption":  {
		"created_time": "1462724530",
		"text": "A true size comparison between my 5 yr old monster cat and our new 8 week old kitten 🐱\n#kittycity #bigvssmall #cats #lillulu #babysavannah #lol #cute #cuddlesesh",
		"from":  {
			"username": "beardgilley90",
			"profile_picture": "https://scontent.cdninstagram.com/t51.2885-19/s150x150/12677517_961393443956882_1140052859_a.jpg",
			"id": "2334123855",
			"full_name": "Evan Pengilley"
		},
		"id": "17847251569116099"
	},
	"user_has_liked": false,
	"id": "1245746113076281966_2334123855",
	"user":  {
		"username": "beardgilley90",
		"profile_picture": "https://scontent.cdninstagram.com/t51.2885-19/s150x150/12677517_961393443956882_1140052859_a.jpg",
		"id": "2334123855",
		"full_name": "Evan Pengilley"
	}
}
```

**Comment by fujiy** [comment]
Thanks. Fixed in #107
